lib/Echidna/Mutator.hs

```diff
@@ -71,3 +71,15 @@ spliceAtRandom :: (LL.ListLike f i, MonadRandom m) => f -> f -> m f
 spliceAtRandom xs1 xs2 = do idx1 <- getRandomR (0, LL.length xs1 - 1) 
                             idx2 <- getRandomR (0, LL.length xs2 - 1)
                             return $ LL.take idx1 xs1 `LL.append` LL.drop idx2 xs2
+
+interleaveAtRandom :: (LL.ListLike f i, MonadRandom m) => f -> f -> m f
+interleaveAtRandom xs1 xs2 = do idx1 <- getRandomR (0, LL.length xs1 - 1) 
+                                idx2 <- getRandomR (0, LL.length xs2 - 1)
+                                return $ LL.take idx1 xs1 `interleaveLL` LL.take idx2 xs2
+
+{- | Takes two lists and combines them interleaving its elements -}
+interleaveLL :: (LL.ListLike f i) => f -> f -> f
```

Comment:
I feel it would be good to have a unit test for this

Reply:
Specific tests for all the array-mutation functions will be great, but I'm not sure how do it. Moreover, this code should rarely change (unless there is a obvious bug), so I'm not sure how useful will be to have unit tests (beyond the first time).